src/main/java/com/exasol/adapter/document/files/GcsRemoteFileFinder.java

```diff
@@ -41,16 +45,23 @@ private Storage buildGcsClient(final GcsConnectionProperties connectionPropertie
         return clientBuilder.build().getService();
     }
 
-    private GoogleCredentials getCredentials(final GcsConnectionProperties connectionProperties) {
-        try (final ByteArrayInputStream inputStream = new ByteArrayInputStream(connectionProperties.getGcKey());) {
-            return GoogleCredentials.fromStream(inputStream);
-        } catch (final IOException exception) {
-            throw new IllegalArgumentException(
-                    ExaError.messageBuilder("E-VS-GCS-9").message("Failed to read Google key definition.")
-                            .mitigation("Please check the value of gcKey.").toString());
+    private Credentials getCredentials(final GcsConnectionProperties connectionProperties) {
+        final byte[] gcKey = connectionProperties.getGcKey();
+        if (Arrays.equals(gcKey, "null".getBytes(StandardCharsets.UTF_8))) {
```

Comment:
Let `getGcKey()` return an `Optional<byte[]>`. This will simplify this condition.